File: ios/src/Bridge/KernelBridge.m
#import "KernelBridge.h"
#include "JSCocoaController.h"
#include "JavascriptCore-dlsym.h"
#include "TWBridgePageRegistry.h"
#include "TWBridgeURLRequestManager.h"

@implementation KernelBridge

+ (void)startWith:(UINavigationController *)root
{
  // Fetch JS symbols
  [JSCocoaSymbolFetcher populateJavascriptCoreSymbols];
    
  // Load iPhone bridgeSupport
  NSString *bundle = [[NSBundle mainBundle] bundlePath];
  [[BridgeSupportController sharedController] loadBridgeSupport:[NSString stringWithFormat:@"%@/iPhone.bridgesupport",  bundle]];
  id c = [JSCocoaController sharedController];
  [c setUseJSLint:NO];
  // Load js libraries
  [c evalJSFile:[NSString stringWithFormat:@"%@/public/assets/scripts/underscore.js", bundle]];
  [c evalJSFile:[NSString stringWithFormat:@"%@/public/assets/scripts/date.js", bundle]];
    
  // Load js bridge
  [c evalJSFile:[NSString stringWithFormat:@"%@/public/assets/scripts/env.js", bundle]];
  [c evalJSFile:[NSString stringWithFormat:@"%@/js/bridge.js", bundle]];
    
  [[TWBridgePageRegistry sharedRegistry] setRoot:root];
  [[TWBridgeURLRequestManager sharedManager] setRoot:root];
}

+ (void)launch:(NSString *)flow
{
  NSString *bundle = [[NSBundle mainBundle] bundlePath];
  id c = [JSCocoaController sharedController];
  [c evalJSString:@"calatrava.app.rootFeature().start()"];
}

@end

File: lib/calatrava/templates/droid/calatrava/src/com/calatrava/bridge/KernelBridge.java
package com.calatrava.bridge;

import java.io.IOException;

public class KernelBridge {

  private AssetRepository assetRepository;
  private RhinoService rhinoService;

  public KernelBridge(AssetRepository assetRepository, RhinoService rhinoService) throws IOException {
    this.assetRepository = assetRepository;
    this.rhinoService = rhinoService;

    //load js libraries
    loadLibrary("calatrava/scripts/underscore.js");

    //load bridge
    loadLibrary("calatrava/scripts/env.js");
    loadLibrary("calatrava/scripts/bridge.js");
  }

  public void loadLibrary(String libraryName) throws IOException {
    rhinoService.load(assetRepository.assetReader(libraryName), libraryName);
  }
}
